src/tracked_struct/tracked_field.rs

```diff
@@ -85,9 +85,10 @@ where
         input: Option<Id>,
         revision: crate::Revision,
     ) -> bool {
-        let current_revision = db.runtime().current_revision();
         let id = input.unwrap();
-        let data = self.struct_map.get(current_revision, id);
+        let data = self
```

Comment:
I don't think I understand salsa enough to make a real assessment but changing the last changed in `maybe_changed_after` seems slightly off to me. 

I also wonder if we'll run into problems with it if the setup is slightly different because, as far as I understand, `maybe_changed_after` is only called from deep verify memo. That means, it would get by passed if all queries only perform a shallow compare. I'm not sure if that matters for tracked fields or if some other mechanism would take over. 

But maybe it is allright because it is kind of similar to what `FunctionIngredient::maybe_changed_after` does.

Reply:
> I also wonder if we'll run into problems with it if the setup is slightly different because, as far as I understand, maybe_changed_after is only called from deep verify memo. That means, it would get by passed if all queries only perform a shallow compare.

Yes, this relates to my question above about whether it's possible for other code paths besides `maybe_changed_after` which call `get` to run into this bug.

But I _think_ it is safe as-is, because any query of low enough durability to possibly re-run in the current revision will also go through deep verify, which will check `maybe_changed_after` on all of its inputs. So a struct that never gets `maybe_changed_after` called on it in a given revision can't be accessed in that revision. I think?

> But maybe it is allright because it is kind of similar to what `FunctionIngredient::maybe_changed_after` does.

Yes, it seems like it is already part of the contract of `maybe_changed_after` that it can validate things to the current revision.